### PR TITLE
Record at native sampling rate.

### PR DESCRIPTION
**Pull request by lduffy-voysis**
Removed createAudioRecorder() static method from utils. Calculating sample rate and buffer size from audioManager.getProperty call.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,13 +1,13 @@
 // Top-level build file where you can add configuration options common to all sub-projects/modules.
 
 buildscript {
-    ext.kotlin_version = '1.2.71'
+    ext.kotlin_version = '1.3.20'
```
[the remark this answers is not in the document]
Comment (oranoceallaigh) [reply build.gradle]: Aye, they should be independent

File: library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt [hunk 2]
```diff
@@ -19,6 +21,10 @@ class AudioRecorderImpl(context: Context,
                         private val player: AudioPlayer = AudioPlayer(context),
                         private var record: AudioRecord? = null,
                         private val executor: Executor = Executors.newSingleThreadExecutor()) : AudioRecorder {
+
+    private val audioManager: AudioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager
```
Comment (oranoceallaigh) [inline library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt]: I was looking around at the docs available for this API and stumbled on people talking about resource leaks related to this API. Have you evaluated these and disregarded them as a problem, or is it something that needs to be looked at more closely?

https://android-review.googlesource.com/c/platform/frameworks/base/+/140481/1
https://stackoverflow.com/questions/20244288/audiomanagergetsystemservicecontext-audio-service-causing-memory-leak

File: library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt [hunk 3]
```diff
@@ -113,4 +119,8 @@ class AudioRecorderImpl(context: Context,
             record = null
         }
     }
+
+    private fun createAudioRecorder(): AudioRecord {
+        return AudioRecord(MediaRecorder.AudioSource.VOICE_RECOGNITION, rate, AudioFormat.CHANNEL_IN_MONO, AudioFormat.ENCODING_PCM_16BIT, size * 4)
```
Comment (oranoceallaigh) [inline library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt]: The buffer sizing makes me a little nervous. You don't care what comes back, just increase it by a factor of 4. I would be more inclined to have some pre-selected buffer sizes chosen per sample rate, but allow a calling application to explicitly state the buffer size it wants to use via configuration.

Either use what the app has explicitly specified _or_:

< 16KHz go for a 2K buffer (== 1K samples per buffer)
>= 16KHz < 44K1Hz go for a 4K buffer (== 2K samples per buffer)
>= 44K1Hz go for an 8K buffer (==4K samples per buffer)

File: build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (lduffy-voysis) [reply build.gradle]: noted. In future i'll put build related things in their own pr.

File: library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt (hunk 2 above)
Comment (lduffy-voysis) [reply library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt]: That could have potentially been a problem. I updated the code just pull the sample rate value from the audioManager and to not hold a reference to it. Code's been refactored and moved to Utils class.

File: library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt (hunk 3 above)
Comment (lduffy-voysis) [reply library/src/main/java/com/voysis/recorder/AudioRecorderImpl.kt]: Updated the code to set the default readBuffer size to 4k and recordBuffer size to 16k. Also made it configurable by the user.

**Pull request merged**